### PR TITLE
[MM-468]: Include number of lines changed in RHS PR list

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
### Summary
Include the number of lines changed in the RHS PR list

#### Existing
Screenshot from 2024-09-03 16-55-42
Screenshot from 2024-09-03 16-55-52
#### Updated
Screenshot from 2024-09-03 16-50-28
Screenshot from 2024-09-03 16-50-41

### How to test
- Connect your mattermost to GitHub
- Create a PR on GitHub with another account and add the GitHub account connected with mattermost as the reviewer
- Open the RHS PR list

### CODE REVIEW

File: server/plugin/graphql/lhs_request.go [hunk 1]
```diff
@@ -19,9 +19,17 @@ const (
 	queryParamAssigneeQueryArg = "assigneeQueryArg"
 )
 
-func (c *Client) GetLHSData(ctx context.Context) ([]*github.Issue, []*github.Issue, []*github.Issue, error) {
+type GithubPRDetails struct {
+	*github.Issue
+	Additions    *githubv4.Int `json:"additions,omitempty"`
+	Deletions    *githubv4.Int `json:"deletions,omitempty"`
+	ChangedFiles *githubv4.Int `json:"changed_files,omitempty"`
```
Comment (raghavaggarwal2308) [inline server/plugin/graphql/lhs_request.go:26]:
```suggestion
	FilesChanged *githubv4.Int `json:"files_changed, omitempty"`
```

File: server/plugin/graphql/lhs_request.go [hunk 2]
```diff
@@ -94,11 +102,11 @@ func (c *Client) GetLHSData(ctx context.Context) ([]*github.Issue, []*github.Iss
 	return resultReview, resultAssignee, resultOpenPR, nil
 }
 
-func getPR(prResp *prSearchNodes) *github.Issue {
+func getPR(prResp *prSearchNodes) *GithubPRDetails {
 	resp := prResp.PullRequest
 	labels := getGithubLabels(resp.Labels.Nodes)
 
-	return newGithubIssue(resp.Number, resp.Title, resp.Author.Login, resp.Repository.URL, resp.URL, resp.CreatedAt, resp.UpdatedAt, labels, resp.Milestone.Title)
+	return newGithubPR(resp.Number, resp.Title, resp.Author.Login, resp.Repository.URL, resp.URL, resp.CreatedAt, resp.UpdatedAt, labels, resp.Milestone.Title, resp.Additions, resp.Deletions, resp.ChangedFiles)
```
Comment (raghavaggarwal2308) [inline server/plugin/graphql/lhs_request.go]: I don't think there is need for creating this function we can directly return the data from here. Please let me know if you have something else in mind.

File: webapp/src/components/sidebar_right/github_items.tsx [hunk 3]
```diff
@@ -227,6 +227,37 @@ function GithubItems(props: GithubItemsProps) {
...
+            const deletions = item?.deletions;
+            const changedFiles = item?.changed_files;
+
+            pullRequestDetails = (
+                <div
+                    className='light'
+                    style={style.pullRequestDetails}
+                >
+                    <FileDiffIcon size={16}/>
+                    {changedFiles != null && (
```
Comment (raghavaggarwal2308) [inline webapp/src/components/sidebar_right/github_items.tsx]:
```suggestion
                    {changedFiles && (
```

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: server/plugin/graphql/lhs_request.go (hunk 1 above)
Comment (Kshitij-Katiyar) [reply server/plugin/graphql/lhs_request.go:26]: cant update the JSON, should we update the var name only?

File: server/plugin/graphql/lhs_request.go (hunk 2 above)
Comment (Kshitij-Katiyar) [reply server/plugin/graphql/lhs_request.go]: the new function is used only once so yes we can directly return the data

File: server/plugin/graphql/lhs_request.go (hunk 1 above)
Comment (raghavaggarwal2308) [reply server/plugin/graphql/lhs_request.go:26]: Why can't we change the json?

Comment (Kshitij-Katiyar) [reply server/plugin/graphql/lhs_request.go:26]: @raghavaggarwal2308 API response is not getting serialized if we update the json

**Review by raghavaggarwal2308 — APPROVED** [review]

**Comment by Kshitij-Katiyar** [comment]
Created a PR on mattermost remote with all these changes hence closed this one

**Pull request closed without merging**